orttraining/orttraining/test/gradient/pass_through_op_test.cc

```diff
@@ -0,0 +1,71 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include <algorithm>
+#include <bitset>
+#include <cmath>
+#include <random>
+#include <thread>
+
+#include "gtest/gtest.h"
+
+#include "test/common/tensor_op_test_utils.h"
+#include "test/util/include/test_random_seed.h"
+#include "orttraining/test/gradient/gradient_op_test_utils.h"
+
+#include "onnx/defs/attr_proto_util.h"
+
+namespace onnxruntime {
+namespace test {
+
+template <typename TTensor, typename TControl>
+void pass_through_multiple_tensors() {
+  std::vector<TControl> control = {10.0f};
+  std::vector<TTensor> input1 = {2.2f, 4.7f, 9.6f};
+  std::vector<TTensor> input2 = {0.6f, 4.3f};
+  OpTester test_passthrough("PassThrough", 1, onnxruntime::kMSDomain);
+  test_passthrough.AddInput<TControl>("control_signal", {}, control);
+  test_passthrough.AddInput<TTensor>("Input1", {static_cast<int64_t>(input1.size())}, input1);
+  test_passthrough.AddInput<TTensor>("Input2", {static_cast<int64_t>(input2.size())}, input2);
+  // We expect input and output are the same.
+  test_passthrough.AddOutput<TTensor>("Output1", {static_cast<int64_t>(input1.size())}, input1);
+  test_passthrough.AddOutput<TTensor>("Output2", {static_cast<int64_t>(input2.size())}, input2);
+  run_provider_specific_optest(test_passthrough);
+}
+
+TEST(PassThrough, PassThroughMultipleFloatTensors) {
+  pass_through_multiple_tensors<float, float>();
+}
+
+TEST(PassThrough, PassThroughMultipleDoubleTensors) {
+  pass_through_multiple_tensors<double, double>();
+}
+
+TEST(PassThrough, PassThroughMultipleHalfTensors) {
```

Comment:
Can we test mixture of float, double, and float16?

Reply:
sounds good, will add